python/looker_sdk/rtl/api_methods.py

```diff
@@ -73,15 +73,20 @@ def __exit__(self, *exc) -> None:
         self.auth.logout()
 
     def _return(self, response: transport.Response, structure: TStructure) -> TReturn:
+        encoding = response.encoding
         if not response.ok:
-            raise error.SDKError(response.value)
+            raise error.SDKError(response.value.decode(encoding=encoding))
         ret: TReturn
         if structure is None:
             ret = None
-        elif structure is str:
+        elif response.response_mode == transport.ResponseMode.BINARY:
             ret = response.value
         else:
-            ret = self.deserialize(response.value, structure)
+            value = response.value.decode(encoding=encoding)
+            if structure is Union[str, bytes] or structure is str:  # type: ignore
```

Comment:
The generated methods pass in this `structure` argument (which is just the method return type) to tell us how to deserialize. In the case of `method.responseIsBoth()` methods which have `structure == Union[str, bytes]` (e.g. `run_url_encoded_query()`) or methods where the return type is just `str` (e.g `create_git_deploy_key()`),  we just return the string as is (since we already decoded the bytes on the line above)

All other `structure` types will get passed to `self.deserialize()`

There is a bug here: the `structure` argument to `self.deserialize(value, structure)` on L89 could theoretically be something like `Union[models.Dashboard, bytes]` (though it doesn't seem like we have anything like that in our Looker API). `self.deserializeI` won't take that type of structure argument so I need to add a check for this case and pull the non-bytes-type out of the `Union` and pass it instead

There might be another bug: I don't see it in our API but I suppose `int`, `float`, and other basic types could be method return types as well and I'm not handling those.

I might just file an issue for these last two potential bugs since this code does work for the Looker API right now.